jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -327,10 +328,18 @@ public String encodeURL(String url)
         if (sessionManager == null)
             return url;
 
-        HttpURI uri = null;
+        URI uri = null;
         if (sessionManager.isCheckingRemoteSessionIdEncoding() && URIUtil.hasScheme(url))
         {
-            uri = new HttpURI(url);
+            try
+            {
+                uri = new URI(url);
```

Comment:
I'm very much against introducing the jdk URI class here.   
I'm going to have a little bit of a play to see what our alternatives are. Standby....

Reply:
Also it is not needed at this point as a URI starting with .. will never have a scheme